command/agent/job_endpoint_test.go

```diff
@@ -1981,29 +1983,150 @@ func TestHTTP_JobDispatch(t *testing.T) {
 
 		// Make the HTTP request
 		req2, err := http.NewRequest(http.MethodPut, "/v1/job/"+job.ID+"/dispatch", buf)
+		must.NoError(t, err)
 		if err != nil {
-			t.Fatalf("err: %v", err)
+			t.Logf("err: %v", err)
 		}
 		respW.Flush()
 
 		// Make the request
 		obj, err := s.Server.JobSpecificRequest(respW, req2)
 		if err != nil {
-			t.Fatalf("err: %v", err)
+			t.Logf("err: %v", err)
 		}
 
 		// Check the response
 		dispatch := obj.(structs.JobDispatchResponse)
 		if dispatch.EvalID == "" {
-			t.Fatalf("bad: %v", dispatch)
+			t.Logf("bad: %v", dispatch)
```

Comment:
We have `must.NotEq(t, "", dispatch.EvalID)` below, which will log if we fail the assertion, so we can remove this whole block. Same applies to the assertion after that.